src/node_dotenv.cc

```diff
@@ -130,50 +135,81 @@ void Dotenv::ParseContent(const std::string_view input) {
 
   while (!content.empty()) {
     // Skip empty lines and comments
+    // Example:
+    //   # This is a comment
+    //
+    //   KEY=value
     if (content.front() == '\n' || content.front() == '#') {
       auto newline = content.find('\n');
       if (newline != std::string_view::npos) {
         content.remove_prefix(newline + 1);
+        // Trim spaces after skipping comments or empty lines to handle
+        // cases where there might be trailing whitespace
+        content = trim_spaces(content);
         continue;
+      } else {
+        // If no newline is found, we've reached the end of content
+        break;
       }
     }
 
-    // If there is no equal character, then ignore everything
-    auto equal = content.find('=');
-    if (equal == std::string_view::npos) {
+    // Find the next equals sign or newline in a single pass
+    // This optimizes the search by avoiding multiple iterations
+    auto pos = content.find_first_of("=\n");
+
+    // If we found nothing or found a newline before equals, the line is invalid
+    if (pos == std::string_view::npos || content[pos] == '\n') {
+      if (pos != std::string_view::npos) {
+        content.remove_prefix(pos + 1);
+        content = trim_spaces(content);
+        continue;
+      }
       break;
     }
 
-    key = content.substr(0, equal);
-    content.remove_prefix(equal + 1);
+    // We found an equals sign, extract the key
+    key = content.substr(0, pos);
+    content.remove_prefix(pos + 1);
     key = trim_spaces(key);
     content = trim_spaces(content);
 
+    // Skip lines with empty keys after trimming spaces
+    // Examples of invalid keys that would be skipped:
+    //   =value
+    //   "   "=value
     if (key.empty()) {
+      auto newline = content.find('\n');
+      if (newline != std::string_view::npos) {
+        content.remove_prefix(newline + 1);
```

Comment:
if `newline` is the last character, this will literally clear the content and we don't need to call trim_spaces afterwards.